modules/spring-config/src/main/java/org/mule/runtime/config/spring/internal/MuleDocumentLoader.java

```diff
@@ -58,7 +60,48 @@ public void handle(short operation, String key, Object data, Node src, Node dst)
     }
   };
 
-  private final DocumentLoader defaultLoader = new DefaultDocumentLoader();
+  /**
+   * JAXP attribute used to configure the schema language for validation.
+   */
+  private static final String SCHEMA_LANGUAGE_ATTRIBUTE = "http://java.sun.com/xml/jaxp/properties/schemaLanguage";
+
+  /**
+   * JAXP attribute value indicating the XSD schema language.
+   */
+  private static final String XSD_SCHEMA_LANGUAGE = "http://www.w3.org/2001/XMLSchema";
+
+  private final DocumentLoader defaultLoader = new DefaultDocumentLoader() {
+
+    protected DocumentBuilderFactory createDocumentBuilderFactory(int validationMode, boolean namespaceAware)
+        throws ParserConfigurationException {
+
+      DocumentBuilderFactory factory = XMLSecureFactories.createDefault().getDocumentBuilderFactory();
+      factory.setNamespaceAware(namespaceAware);
+
+      if (validationMode != XmlValidationModeDetector.VALIDATION_NONE) {
+        factory.setValidating(true);
+        if (validationMode == XmlValidationModeDetector.VALIDATION_XSD) {
+          // Enforce namespace aware for XSD...
+          factory.setNamespaceAware(true);
+          try {
+            factory.setAttribute(SCHEMA_LANGUAGE_ATTRIBUTE, XSD_SCHEMA_LANGUAGE);
+          } catch (IllegalArgumentException ex) {
+            ParserConfigurationException pcex = new ParserConfigurationException(
+                                                                                 "Unable to validate using XSD: Your JAXP provider ["
+                                                                                     + factory +
+                                                                                     "] does not support XML Schema. Are you running on Java 1.4 with Apache Crimson? "
+                                                                                     +
+                                                                                     "Upgrade to Apache Xerces (or Java 1.5) for full XSD support.");
```

Comment:
The code is copied from Spring but changing the factory construction, I could but it might be incompatible if something depends on the error message.